### ISSUE TITLE
Error when Construct Props are not used within the Constructor

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#166]
### Feature Description

CDK Construct Props are expected to always be used within the Construct, so error when all properties of Props are not used within the Construct

### Motivation

To prevent defining unnecessary Props properties

### Invalid Code Examples

```ts
export type MyConstructProps = {
   bucketName: string // ❌ `bucketName` property never used in Construct
}

class MyConstruct extends Construct {
  constructor(scope: Construct, id: string, props: MyConstructProps) {
    super(scope, id);
    new Bucket(this, "MyBucket");
  }
}
```

### Valid Code Examples

```ts
export type MyConstructProps = {
   bucketName: string // ✅ `bucketName` property used in Construct
}

class MyConstruct extends Construct {
  constructor(scope: Construct, id: string, props: MyConstructProps) {
    super(scope, id);
    new Bucket(this, "MyBucket", {
      bucketName: props.bucketName
    });
  }
}
```

### Rule Type

suggestion (improve code)

### Implementation Details

- [x] This rule is related to AWS CDK best practices
- [x] I'm willing to submit a PR for this feature

**Comment by ren-yamanashi** [comment]
@claude 

Please provide more detail on the content of the issue
